### PR TITLE
Add support for Ecto 3.10

### PR DESCRIPTION
**Pull request by fredwu**
Based on #101 

Now that Ecto 3.10 has been released for a little while, let's make `ecto_sqlite3` compatible with it. This PR is build on top of the amazing work @jeregrine has done in #101.

### CODE REVIEW

File: lib/ecto/adapters/sqlite3/connection.ex [hunk 1]
```diff
@@ -332,11 +332,11 @@ defmodule Ecto.Adapters.SQLite3.Connection do
...
 
-  defp build_explain_query(query, :query_plan) do
+  def build_explain_query(query, :query_plan) do
     IO.iodata_to_binary(["EXPLAIN QUERY PLAN ", query])
   end
 
-  defp build_explain_query(query, :instructions) do
+  def build_explain_query(query, :instructions) do
     IO.iodata_to_binary(["EXPLAIN ", query])
   end
```
Comment (warmwaffles) [inline lib/ecto/adapters/sqlite3/connection.ex:341]: Can both of these be left private?

Comment (fredwu) [reply lib/ecto/adapters/sqlite3/connection.ex:341]: These changes are from #101, cc @jeregrine

Comment (jeregrine) [reply lib/ecto/adapters/sqlite3/connection.ex:341]: I don't remember exactly why I made them public, but I also don't feel like there is any reason to have private functions in elixir lol. Not like we are hiding data or encapsulating an API. Almost all end users will never use this api.

**Comment by warmwaffles** [comment]
@fredwu can you rebase off of latest `main`?

File: lib/ecto/adapters/sqlite3/connection.ex (hunk 1 above)
Comment (warmwaffles) [reply lib/ecto/adapters/sqlite3/connection.ex:341]: I don't really agree with that, but it's a topic for a different day.

**Comment by fredwu** [comment]
@warmwaffles Rebasing was a mess, so I cherry-picked and created #108.

**Pull request closed without merging**